lib/ansible/modules/database/postgresql/postgresql_facts.py

```diff
@@ -0,0 +1,597 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Andrew Klychkov (@Andersson007) <aaklychkov@mail.ru>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: postgresql_facts
+short_description: Gathers facts about remote PostgreSQL servers.
+description:
+   - Gathers facts about remote PostgreSQL servers.
+version_added: "2.8"
+options:
+  include_subset:
+    description:
+      - Limit collected subsets by comma separated list. Allowable values are C(version),
+        C(databases), C(settings), C(tablespaces), C(languages), C(roles), C(namespaces),
+        C(replications), C(repl_slots), C(extensions). By default, collects all subsets.
+        You can use shell-style (fnmatch) wildcard to pass groups of values (see Examples).
```

Comment:
We tend to add parameters type for each parameter, so it ends up in the documentation.
```suggestion
        You can use shell-style (fnmatch) wildcard to pass groups of values (see Examples).
      type: list
```
Do this for all parameters.

Reply:
This is a string that's split into code to a list. Is it necessary to implement this as a list?

Reply:
Ok, I'll change them to lists. I've just seen your next suggestions.

Reply:
If you make it a list, and people provide a comma-separated string it will do the split for you.
So it will be backward compatible by default.